File: e2e/cypress-javascript/interactions/steps/NavigationStep.js
const { step } = require('@/decorators/step-decorator');

class NavigationStep {
  constructor() {
    this.openMainPage = step('Navigate to main page', () => cy.visit('/'));
    this.refresh = step('Refresh page', () => cy.reload());
  }
}

module.exports = { NavigationStep };

File: e2e/capybara-ruby/interactions/steps/navigation_step.rb
# frozen_string_literal: true

require_relative '../../decorators/step_decorator'
require_relative '../../support/capybara_helpers'

class NavigationStep
  extend StepDecorator::Methods
  include CapybaraHelpers

  step 'Navigate to main page'
  def open_main_page
    visit('/')
  end

  step 'Refresh page'
  def refresh
    page.driver.refresh
  end
end
